### PR TITLE
Improve CoinJoin Description

### PR DESCRIPTION
**Pull request by nopara73**
Extend CoinJoin explanation with currently operational implementations and their descriptions.  
Resolves: https://github.com/bitcoin-dot-org/bitcoin.org/issues/2785

### CODE REVIEW

File: _data/devdocs/en/guides/contracts.md [hunk 1]
```diff
@@ -210,20 +210,20 @@ all under an Apache license.
 
 {% autocrossref %}
 
-Alice is concerned about her privacy.  She knows every transaction gets
-added to the public block chain, so when Bob and Charlie pay her, they
+Alice is concerned about her privacy. She knows every transaction gets
+added to the public Blockchain, so when Bob and Charlie pay her, they
```
Comment (harding) [inline _data/devdocs/en/guides/contracts.md]: Unless it's changed since I last contributed here, Bitcoin.org uses the original Nakamoto-style spelling (and capitalization) for block chain (two words, common noun).  Last time this was discussed, I think we actually settled it with a coin flip to avoid bikeshedding.

File: _data/devdocs/en/guides/contracts.md [hunk 2]
```diff
@@ -210,20 +210,20 @@ all under an Apache license.
...
 can each easily track those satoshis to learn what Bitcoin
 addresses she pays, how much she pays them, and possibly how many
 satoshis she has left.
 
-Alice isn't a criminal, she just wants plausible deniability about
+Alice is not a criminal, she just wants plausible deniability about
```
Comment (harding) [inline _data/devdocs/en/guides/contracts.md]: This is an unnecessary change.  The documentation uses contractions quite extensively.

File: _data/devdocs/en/guides/contracts.md [hunk 3]
```diff
@@ -210,20 +210,20 @@ all under an Apache license.
...
 where she has spent her satoshis and how many she has left, so she
-starts up the Tor anonymity service on her computer and logs into an
+starts up the Tor anonymity network on her computer and logs into an
```
Comment (harding) [inline _data/devdocs/en/guides/contracts.md]: The previous phrasing was correct.  Tor is an anonymity network, but the program you start on your computer is a service.

File: _data/devdocs/en/guides/contracts.md [hunk 4]
```diff
@@ -210,20 +210,20 @@ all under an Apache license.
...
 IRC chatroom as "AnonGirl."
 
-Also in the chatroom are "Nemo" and "Neminem."  They collectively
+"Nemo" and "Neminem" are Also in the chatroom. They collectively
```
Comment (harding) [inline _data/devdocs/en/guides/contracts.md]: This change is weird, and the capitalization in the middle of the sentence is ungrammatical.

File: _data/devdocs/en/guides/contracts.md [hunk 5]
```diff
@@ -276,14 +276,30 @@ to the separate merchant addresses so nobody will be able to figure
...
+[JoinMarket](https://github.com/JoinMarket-Org/) and
+[Wasabi Wallet](http://wasabiwallet.io) are the operational CoinJoin implementations
+for Bitcoin.
+
+JoinMarket style CoinJoins differ from the above described scheme by splitting
+the participants into two sections: market makers and market takers.
+Market makers are publishing their CoinJoin intentions to an IRC
+room and waiting for market takers to take their offers. When a taker comes along,
+it selects a set of makers and creates a shared transaction with them,
+while also paying a small fee. Unlike the above described sheme, this happens automatically.
```
Comment (harding) [inline _data/devdocs/en/guides/contracts.md]: s/sheme/scheme/

File: _data/devdocs/en/guides/contracts.md [hunk 6]
```diff
@@ -276,14 +276,30 @@ to the separate merchant addresses so nobody will be able to figure
...
+Since they would have had to pay a transaction fee to make their purchases
+anyway, AnonGirl and her co-spenders do not pay anything extra---but
 because they reduced overhead by combining multiple transactions, saving
 bytes, they may be able to pay a smaller aggregate transaction fee,
 saving each one of them a tiny amount of satoshis.
 
-**Resource:** An alpha-quality (as of this writing) implementation of decentralized
-CoinJoin is [CoinMux](http://coinmux.com/), available under the Apache
-license.
+**Current Working Implementations:** As of today, in 1018,
```
Comment (harding) [inline _data/devdocs/en/guides/contracts.md]: s/1018/2018/, unless you have a time machine.

File: _data/devdocs/en/guides/contracts.md (hunk 4 above)
Comment (nopara73) [reply _data/devdocs/en/guides/contracts.md]: F*ck, I was sure this time that my English intuition is finally right about something:)

**Comment by harding** [comment]
Untested ACK daed3be3a83aa55935df53f7fe134de976269731

**Comment by wbnns** [comment]
This is scheduled to be merged on Friday, December 21st.
